pypdf/generic/_files.py

```diff
@@ -25,29 +41,124 @@ def __init__(self, name: str, pdf_object: DictionaryObject) -> None:
+                NameObject(FileSpecificationDictionaryEntries.F): create_string_object(name),
+                NameObject(FileSpecificationDictionaryEntries.EF): ef_entry,
+            }
+        )
+
+        # Add to the catalog's names tree
+        if CA.NAMES not in writer._root_object:
+            writer._root_object[NameObject(CA.NAMES)] = writer._add_object(DictionaryObject())
+
+        names_dict = cast(DictionaryObject, writer._root_object[CA.NAMES])
+        if "/EmbeddedFiles" not in names_dict:
+            embedded_files_names_dictionary = DictionaryObject(
+                {NameObject(CA.NAMES): ArrayObject()}
+            )
+            names_dict[NameObject("/EmbeddedFiles")] = writer._add_object(embedded_files_names_dictionary)
+        else:
+            embedded_files_names_dictionary = cast(DictionaryObject, names_dict["/EmbeddedFiles"])
+
+        # Add the name and filespec to the names array
+        names_array = cast(ArrayObject, embedded_files_names_dictionary[CA.NAMES])
+        names_array.extend([create_string_object(name), filespec])
+
+        # Return an EmbeddedFile instance
+        return cls(name=name, pdf_object=filespec)
+
     @property
     def alternative_name(self) -> str | None:
         """Retrieve the alternative name (file specification)."""
         for key in [FileSpecificationDictionaryEntries.UF, FileSpecificationDictionaryEntries.F]:
             # PDF 2.0 reference, table 43:
             #   > A PDF reader shall use the value of the UF key, when present, instead of the F key.
             if key in self.pdf_object:
-                return cast(str, self.pdf_object[key].get_object())
+                value = self.pdf_object[key].get_object()
+                if not is_null_or_none(value):
+                    return cast(str, value)
         return None
 
+    @alternative_name.setter
+    def alternative_name(self, value: str | TextStringObject | None) -> None:
+        """Set the alternative name (file specification)."""
+        if value is None:
+            # Set UF key to null
+            self.pdf_object[NameObject(FileSpecificationDictionaryEntries.UF)] = NullObject()
+        elif isinstance(value, PdfObject):
+            self.pdf_object[NameObject(FileSpecificationDictionaryEntries.UF)] = value
+        else:
+            from pypdf.generic import create_string_object  # noqa: PLC0415
+            self.pdf_object[NameObject(FileSpecificationDictionaryEntries.UF)] = create_string_object(value)
+
     @property
     def description(self) -> str | None:
         """Retrieve the description."""
-        return self.pdf_object.get(FileSpecificationDictionaryEntries.DESC)
+        value = self.pdf_object.get(FileSpecificationDictionaryEntries.DESC)
+        if is_null_or_none(value):
+            return None
+        return value
+
+    @description.setter
+    def description(self, value: str | TextStringObject | None) -> None:
```

Comment:
This looks odd. As in the other more specific APIs, I would suggest to use `value: TextStringObject | None` instead. I think we can expect the user to call `create_string_object(value)` himself (especially when we document it).

`isinstance(value, PdfObject)` looks strange as well - we already know that the only `PdfObject` which is allowed due to the type is a `TextStringObject`.